lib/KBaseReportPy/utils/report_utils.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+from file_utils import validate_paths, fetch_or_upload_files
+from uuid import uuid4
+
+""" Utilities for creating reports using DataFileUtil """
+
+
+def create_report(ctx, params, dfu):
+    """
+    Create a simple report
+    :param ctx: context dict passed into KBaseReportImpl#create
+    :param params: see utils/validation_utils
+    :param dfu: instance of DataFileUtil
+    :returns: report data
+    """
+    report_name = "report_" + str(uuid4())
+    workspace_id = _get_workspace_id(dfu, params)
+    # Set default empty values for various Report parameters
+    report_data = {
+        'objects_created': [],
+        'warnings': [],
+        'file_links': [],
+        'html_links': [],
+        'direct_html': '',
+        'direct_html_link_index': 0
+    }
+    report_data.update(params['report'])
+    save_object_params = {
+        'id': workspace_id,
+        'objects': [{
+            'type': 'KBaseReport.Report',
+            'data': report_data,
+            'name': report_name,
+            'meta': {},
+            'hidden': 1,
+            'provenance': ctx['provenance']
+        }]
+    }
+    obj = dfu.save_objects(save_object_params)[0]
```

Comment:
I usually wrap calls in try/catch and log the "server side" exception, because otherwise it gets lost. See kb_quast for an example.